test/tstDetailsTreeNodeLabeling.cpp

```diff
@@ -0,0 +1,196 @@
+  }                                                                            \
+  template <>                                                                  \
+  KOKKOS_FUNCTION auto                                                         \
+  ArborX::Details::HappyTreeFriends::getRightChild<MockBVH<MEMORY_SPACE>>(     \
+      MockBVH<MEMORY_SPACE> const &x, int i)                                   \
+  {                                                                            \
+    return x.children_(i).second;                                              \
+  }
+
+HAPPY_TREE_FRIENDS_GET_CHILDREN_SPECIALIZATION(
+    Kokkos::DefaultExecutionSpace::memory_space)
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP) ||               \
+    defined(KOKKOS_ENABLE_SYCL) || defined(KOKKOS_ENABLE_OPENMPTARGET)
+static_assert(!std::is_same<Kokkos::DefaultExecutionSpace::memory_space,
+                            Kokkos::HostSpace>::value,
+              "");
+HAPPY_TREE_FRIENDS_GET_CHILDREN_SPECIALIZATION(Kokkos::HostSpace)
+#endif
+
+namespace Test
+{
+template <class ExecutionSpace>
+auto findParents(ExecutionSpace const &exec_space,
+                 std::vector<Kokkos::pair<int, int>> const &children_host)
+{
+  auto children = toView<ExecutionSpace>(children_host, "Test::children");
+
+  auto const n = children.extent(0) + 1;
+  Kokkos::View<int *, ExecutionSpace> parents(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing, "Test::parents"),
+      2 * n - 1);
+  Kokkos::deep_copy(parents, -1);
+
+  using MemorySpace = typename ExecutionSpace::memory_space;
+  ArborX::Details::findParents(exec_space, MockBVH<MemorySpace>{children},
+                               parents);
+  return Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, parents);
+}
+
+#define ARBORX_TEST_FIND_PARENTS(exec_space, children, ref)                    \
+  BOOST_TEST(Test::findParents(exec_space, children) == ref,                   \
+             boost::test_tools::per_element())
+
+template <class ExecutionSpace>
+auto reduceLabels(ExecutionSpace const &exec_space,
+                  std::vector<int> const &parents_host,
+                  std::vector<int> const &labels_host)
+{
+  auto labels = toView<ExecutionSpace>(labels_host, "Test::labels");
+  auto parents = toView<ExecutionSpace>(parents_host, "Test::parents");
+
+  ArborX::Details::reduceLabels(exec_space, parents, labels);
+
+  return Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, labels);
+}
+
+#define ARBORX_TEST_REDUCE_LABELS(exec_space, parents, labels, ref)            \
+  BOOST_TEST(Test::reduceLabels(exec_space, parents, labels) == ref,           \
+             boost::test_tools::per_element())
+
+} // namespace Test
+
+BOOST_AUTO_TEST_SUITE(TreeHelpers)
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(find_parents, DeviceType, ARBORX_DEVICE_TYPES)
+{
+
+  using ExecutionSpace = typename DeviceType::execution_space;
+  ExecutionSpace exec_space;
+
+  // [0]--*--
+  //     / \
```

Comment:
We get warnings for the last `\` in a comment since that results in line continuation.

Reply:
Thx.  Fixed.